lib/src/utf8.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library utf8;
+
+import 'dart:convert';
+import 'dart:ffi';
+import 'dart:typed_data';
+
+/// [Utf8] implements conversion between Dart strings and null-termianted
+/// Utf8-encoded "char*" strings in C.
+class Utf8 extends Struct<Utf8> {
```

Comment:
Given this class has only static members, why should it inherit from `Struct<Utf8>` and why is it not abstract?

It could be just two top-level methods:

```dart
Pointer<Uint8> toNullTerminatedCString(String s) => ...;
String fromNullTerminatedCString(Pointer<Uint8> s) => ...;
```

Reply:
The reason to inherit from `Struct` is that we can make a more precise type `Pointer<Utf8>` which is more informative than `Pointer<Uint8>`. It also allows writing native function signatures which have the argument type `Pointer<Utf8>`.

Reply:
What does `sizeof<Utf8>()` return? Is it 0? So `Pointer<Utf8>` will be a pointer to a 0-length struct?

That means it's basically just a nominal type representing the address and, depending on how function signatures are generated, might require users to cast this opaque pointer to `Pointer<Int8>` for callbacks or return values.

Maybe it's worth mentioning that in the class description.

Reply:
There is no need to cast it in function signatures. I've added a comment.